From cb0ed5007c057943dcf786301a2729892fa5a54a Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 12 Aug 2018 07:24:58 +0300
Subject: [PATCH] board: gaudi is a new ARM64 SoC Fixup

1) update makefile to use correct defconfig
2) remove HL_VARS_FILE at distclean
3) Since our sed replacment is not good enough start with local inittab
	TBD: improve sed and remove local inittab
4) Since we remove HL_VARS_FILE during distclean make sure install behave like
   build goal in term of recursive make to insure HL_VARS_FILE is exist
---

diff --git a/Makefile b/Makefile
index 91be1bd..9ae4b6c 100644
--- a/Makefile
+++ b/Makefile
@@ -105,8 +105,12 @@
 dali-pldm-install: DALI_INSTALL_HOST_IP ?= 192.168.10.45
 dali-asic-install: DALI_INSTALL_LAB_IP ?= 192.168.10.109
 dali-asic-install: DALI_INSTALL_HOST_IP ?= 192.168.10.134
-.PHONY: dali-install gaudi-install
-dali-install gaudi-install:
+dali-onix-install gaudi-onix-install: dali-onix-target
+dali-pldm-install: dali-pldm-target
+dali-asic-install: dali-asic-target
+
+.PHONY: hl-install
+hl-install:
 	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
@@ -121,6 +125,12 @@
 	-sshpass -p123456        scp $(LINUX_BASE)/vmlinux			root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	-sshpass -p123456        scp -r $(UBOOT_XMPL)/$(HL_DEV)*.bin        	root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
 	-sshpass -p123456        scp ${DSV_BASE}/dsv.bin			root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+
+
+.PHONY: dali-install gaudi-install
+dali-install gaudi-install:
+	@$(MAKE) hl-install DALI_INSTALL_HOST_IP=$(DALI_INSTALL_HOST_IP) DALI_INSTALL_LAB_IP=$(DALI_INSTALL_LAB_IP)
+
 gaudi-onix-install: gaudi-install
 dali-onix-install dali-pldm-install dali-asic-install: dali-install
 dali-onix-install gaudi-onix-install:
@@ -133,7 +143,8 @@
 ASIC_BIN_DIR ?= /tools/hl_shared_dir/production/embedded/head
 ${ASIC_BIN_DIR}:
 	@mkdir -p $@
-dali-asic-install: ${ASIC_BIN_DIR}
+.PHONY: hl-asic-install
+hl-asic-install: $(ASIC_BIN_DIR)
 	-cp $(IMAGES_DIR)/$(HL_DEV)-fit.itb $</
 	-cp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin $</
 	-cp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin $</
@@ -142,16 +153,17 @@
 ifneq ($(notdir $(realpath $(ASIC_BIN_DIR))), head)
 	ln -nfs $$(basename $(ASIC_BIN_DIR)) $$(dirname $(ASIC_BIN_DIR))/latest
 endif
+dali-asic-install:
+	@$(MAKE) hl-asic-install
 
 .PHONY: dali-distclean gaudi-distclean
-gaudi-distclean: hl-gaudi
-dali-distclean: hl-dali
-dali-distclean gaudi-distclean: $(OUTPUT_DIR)/.config
-	@$(MAKE) uboot-dirclean linux-dirclean armcp-dirclean	\
-		 spdlog-dirclean specs-dirclean specs_external-dirclean \
-		 host-uboot-test-hooks-dirclean host-tbot-dirclean \
-		 thermal-dirclean preboot-dirclean host-fw_loader-dirclean
-	@rm -f $(HL_VARS_FILE)
+dali-distclean gaudi-distclean:
+	@rm  -f $(HL_VARS_FILE)
+	@cd $(OUTPUT_DIR)/build && \
+	rm -rf uboot-custom linux-custom armcp-custom \
+	rm -rf spdlog-custom specs-custom specs_external-custom \
+	rm -rf host-uboot-test-hooks-custom host-tbot-custom \
+	rm -rf thermal-custom preboot-custom host-fw_loader-custom
 
 .PHONY: dali-onix-test dali-pldm-test dali-asic-test
 dali-onix-test: dali-onix-tools-install
